Unrecognized field "currentItemId" on MediaStatus

haku opened the pull request:
When trying to play a MP3 file through a Chromecast Audio,
the file starts paying,
but `chromecast.getMediaStatus()` seems to break with this:
```
12:15:19.150 [  Thread-2]      s.l.c.api.v2.Channel W Error while reading: Unrecognized field "currentItemId" (Class su.litvak.chromecast.api.v2.MediaStatus), not marked as ignorable
 at [Source: java.io.StringReader@b71498; line: 1, column: 826] (through reference chain: su.litvak.chromecast.api.v2.MediaStatus["currentItemId"])
```

The docs ( https://developers.google.com/cast/docs/reference/receiver/cast.receiver.media.MediaStatus ) describe this field:

"currentItemId
(number or undefined)
ID of this media item (the item that originated the status change)."

I guess it is something to do with playlists, but since the rest of the library currently does not know about `ItemId`, its not much use on its own.
So for now adding `@JsonIgnoreProperties(ignoreUnknown = true)` to `MediaStatus` works around the issue and makes it less brittle to API changes.

p.s. Thanks for creating this library - it looks like just what I need for my next project. :D

vitalidze commented:
Thanks for pull request. Actually I don't like approach of ignoring missing fields. I suggest to just add it per that specification instead and put the ignoreMissing setting to the values which can be `undefined` per specifications. What do you think?

haku commented:
Additional fields added as requested. :)
I can not see anything in the Jackson API for `ignoreMissing`, but it seems to already be setting null for missing fields.  Or at least the existing unit tests still pass.

vitalidze commented:
Thank you. I don't see neither `items` nor `repeatMode` in specification. Also there are following fields from specification which are missing both in yours and original (current mine) implementations:

* activeTrackIds
* loadingItemId
* preloadedItemId
* customData

I suggest to:

1) Add all missing fields from specification
2) Do not use primitive types for the objects that can be `undefined`. For example use `Integer` instead of `int` for the `currentItemId` field.
3) For `customData` field use `Map` type, so any custom object will be mapped into a `Map`, where keys are properties of the object.
4) Write unit tests that will check proper conversion.

Can you please do it?

haku commented:
I have added a unit test based on the responses I am seeing from my ChromeCast Audio.  As you say, some fields are not in the spec, but they are being returned. :(
I added the fields activeTrackIds, customData, loadingItemId, preloadedItemId, but can not add a realistic unit test as I have not seem them in any real response yet.

vitalidze commented:
Merged. Thank you. I will try to fix issues from #12 by merging this code with #13 and add corresponding tests. Then I think I will release 0.0.7.

vitalidze commented:
In #12 I have an example of `MediaStatus` coming from Pandora. It has the `BUFFERED` `streamType` and in [docs](https://developers.google.com/cast/docs/reference/receiver/cast.receiver.media#.StreamType) all values are in upper case. But in your test case it is in lower case. Are you sure it is correct?

vitalidze commented:
FYI v. `0.0.7` is available in maven central.

vitalidze commented:
@haku can you please answer [my question](https://github.com/vitalidze/chromecast-java-api-v2/pull/14#issuecomment-150836704) ? It stops from merging #17 and fixing #15

haku commented:
Sorry for not answering before.
Just ran my test again against my ChromeCast Audio with firmware 1.16.43955, this is exactly what I see logged (except I changed the media URL):
```
20:47:30.034 [  Thread-2]      s.l.c.api.v2.Channel D  <-- {"type":"MEDIA_STATUS","status":[{"mediaSessionId":1,"playbackRate":1,"playerState":"BUFFERING","currentTime":0,"supportedMediaCommands":15,"volume":{"level":1,"muted":false},"media":{"contentId":"http://192.168.1.6:8192/audio-thing.mp3","contentType":"audio/mpeg","duration":95.294667,"streamType":"buffered"},"currentItemId":1,"items":[{"itemId":1,"media":{"contentId":"http://192.168.1.6:8192/audio-thing.mp3","contentType":"audio/mpeg","duration":95.294667,"streamType":"buffered"},"autoplay":true,"customData":{"payload":{"title:":"Toad Cast","thumb":null}}}],"repeatMode":"REPEAT_OFF"}],"requestId":5}
```

vitalidze commented:
Thank you for answering. Then I guess the `streamType` should be handled in a case-insensitive manner.
